app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -360,4 +343,33 @@ private int getMemoryIncreaseForGC() {
       return 0;
     }
   }
+
+  private void suggestExecutorInstances() {
+    if (suggestedCore != lastRunExecutorCore) {
+      suggestExecutorInstancesBasedOnCore();
+    } else {
+      suggestedExecutorInstances = lastRunExecutorInstances;
+    }
+    if (suggestedExecutorInstances > CLUSTER_DEFAULT_DYNAMIC_ALLOCATION_MAX_EXECUTOR) {
+      suggestedExecutorInstances = CLUSTER_DEFAULT_DYNAMIC_ALLOCATION_MAX_EXECUTOR;
+    }
+    //job fails if max executors is less than executor instances
+    if (lastRunDynamicAllocationMaxExecutors != null
+        && suggestedExecutorInstances > lastRunDynamicAllocationMaxExecutors) {
+      suggestedExecutorInstances = lastRunDynamicAllocationMaxExecutors;
+    }
+  }
+
+  private void suggestExecutorInstancesBasedOnCore() {
+    Integer lastRunActualExecutorInstances = lastRunExecutorInstances;
```

Comment:
Can this be null other than the case where Dr.Elephant does not have this heuristic due to some issue in processing the job?